packages/@ember/-internals/runtime/lib/mixins/array.d.ts

```diff
@@ -0,0 +1,103 @@
+type Value<T, K extends string> = K extends keyof T ? T[K] : unknown;
+
+export type NatifiedArray<T> = T[] & NativeArray<T>;
+
```

Comment:
This is the same as `NativeArray` itself.

```suggestion
```

Reply:
It isn't the same right now, but if we get it defined correctly, I agree that it should be.